Add jsdoc for documentation

Robdel12 opened the pull request:
Inside of a function use `C-c i` to get this effect:
![2017-05-07 23 22 17](https://cloud.githubusercontent.com/assets/2072894/25834085/408f4f24-343b-11e7-948a-34457ffd8167.gif)

Robdel12 committed:
Add jsdoc for documentation

cowboyd commented on frontmacs-javascript.el:
```diff
@@ -17,4 +18,10 @@
+(add-hook 'js2-mode-hook
+           #'(lambda ()
+               (define-key js2-mode-map "\C-ci" 'js-doc-insert-function-doc)
+               (define-key js2-mode-map "@" 'js-doc-insert-tag)))
```
The first sharp quote is redundant, but the other functions could use them:

```elisp
(add-hook 'js2-mode-hook
           (lambda ()
               (define-key js2-mode-map "\C-ci" #'js-doc-insert-function-doc)
               (define-key js2-mode-map "@" #'js-doc-insert-tag)))
```

http://endlessparentheses.com/get-in-the-habit-of-using-sharp-quote.html is a great read on when to use sharp quote and when not.

Robdel12 committed:
Use quote sharp right

cowboyd committed:
remap the key binding for jsdoc to C-c C-r d

We already have a prefix for "advanced" features, and the "C-c i"
prefix might be useful globally down the road.
